diesel/src/type_impls/primitives.rs

```diff
@@ -19,6 +19,11 @@ mod foreign_impls {
     #[diesel(sql_type = Bool)]
     struct BoolProxy(bool);
 
+    #[derive(AsExpression, FromSqlRow)]
+    #[diesel(foreign_derive)]
+    #[cfg_attr(feature = "postgres_backend", diesel(sql_type = crate::sql_types::CChar))]
```

Comment:
These impl is also postgres specific. We need to gate the `AsExpression` derive as well on `feature = "postgres_backend".